llvm/lib/Transforms/Vectorize/LoopVectorizationPlanner.h

```diff
@@ -331,6 +344,8 @@ class LoopVectorizationPlanner {
   /// Profitable vector factors.
   SmallVector<VectorizationFactor, 8> ProfitableVFs;
 
+  SmallVector<PartialReductionChain> PartialReductionChains;
```

Comment:
Is this needed in the scope of LVP or can be limited to constructing the VPlans?

Reply:
It looks like this is cleaner than passing it between functions with parameters. And the functions that use it are all part of `LoopVectorizationPlanner` anyway.

Reply:
I think this can be moved to the cost model, which is available to both the LoopVectorizationPlanner and the VPRecipeBuilder. It can be handled similarly to CallInsts, with a map between the relevant instruction (loop exit instr from the recurrence descriptor?) combined with VF, and a struct containing the relevant details. This also means you don't have to pass the Planner into the cost model.

Reply:
Done. I couldn't include VF in the map since it's not available at the point of creating the recipe and it would be needed to fetch the chain from the map.